ImageFeed/Services/OAuth2Service.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  OAuth2Service.swift
+//  ImageFeed
+//
+//  Created by Andrey Sysoev on 16.12.2022.
+//
+
+import Foundation
+
+fileprivate let UnsplashAuthorizeURLString = "https://unsplash.com/oauth/token"
+
+final class OAuth2Service {
+    private enum OAuthError: Error {
+        case codeError, decodeError
+    }
+    
+    func fetchAuthToken(code: String, completion: @escaping (Result<String, Error>) -> Void) {
+        let url = URL(string: UnsplashAuthorizeURLString)!
+        var request = URLRequest(url: url)
+        
+        let payload: [String: String] = [
+            "client_id": AccessKey,
+            "client_secret": SecretKey,
+            "redirect_uri": RedirectURI,
+            "code": code,
+            "grant_type": "authorization_code"
+        ]
+        
+        request.httpMethod = "POST"
+        request.httpBody = try? JSONSerialization.data(withJSONObject: payload)
+        
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        
+        let task: URLSessionDataTask = URLSession.shared.dataTask(with: request) { data, response, error in
+            if let error {
+                completion(.failure(error))
+                return
+            }
+            
+            if
+                let response = response as? HTTPURLResponse,
+                response.statusCode < 200 || response.statusCode > 299
```

Comment:
Тут я бы использовала все таки код 300
Исхожу из того, что как-то условно принято делить коды на разные категории по сотням. И встретить более длинное `>=` но зато понятное `300` – гораздо предсказуемее что ли, чем то что есть сейчас – приходится внимательно вчитываться в условие и число, чтобы убедиться что нет ошибки. 
Иногда в погоне за локаничностью, можно чуть переборщить и написать логически верное выражение, но семантика за ним – потеряется. 

Надеюсь, получилось объяснить :) 
```suggestion
                response.statusCode < 200 || response.statusCode >= 300
```